services/galley/src/Galley/API/Update.hs

```diff
@@ -98,36 +103,62 @@ unblockConv (usr ::: conn ::: cnv) = do
 joinConversation :: UserId ::: ConnId ::: ConvId ::: JSON -> Galley Response
 joinConversation (zusr ::: zcon ::: cnv ::: _) = do
     c <- Data.conversation cnv >>= ifNothing convNotFound
+    when (Data.isConvDeleted c) $ do
+        Data.deleteConversation cnv
+        throwM convNotFound
```

Comment:
Presumably we can make this transparent by moving it into `Data.conversation` and return `Nothing` in those cases. I would suggest to do that little refactoring later though.